include/dali/imgcodec/image_format.h

```diff
@@ -71,32 +71,34 @@ class DLL_PUBLIC ImageFormat {
   /**
    * @brief Gets a pointer to the image parser
    */
-  const ImageParser* Parser() const;
+  const ImageParser *Parser() const;
 
   /**
    * @brief Returns a string representing the name of the format
    */
   const std::string& Name() const;
 
   /**
-   * @brief Returns a set of decoders associated with this format
+   * @brief Returns a set of decoder factories associated with this format
    */
-  span<ImageDecoder* const> Decoders() const;
+  span<ImageDecoderFactory *const> Decoders() const;
 
   /**
    * @brief Registers a new decoder associated with this format, with a set priority
    *
-   * @param decoder   A decoder
+   * A decoder is registered via a factory.
+   *
+   * @param factory   A decoder factory
    * @param priority  A float representing the priority of this codec.
    *                  The lower the number, the higher priority the codec has.
    */
-  void RegisterDecoder(std::shared_ptr<ImageDecoder> decoder, float priority);
+  void RegisterDecoder(std::shared_ptr<ImageDecoderFactory> factory, float priority);
 
  private:
   std::string name_;
   std::shared_ptr<ImageParser> parser_;
```

Comment:
Dont get why it is shared_ptr

Reply:
Yeah, in principle, it doesn't need to be - but what would be the alternative? `unique_ptr` + `bool own`?